web/src/drafts.js

```diff
@@ -308,8 +306,6 @@ export function restore_draft(draft_id) {
     compose_fade.clear_compose();
     compose.clear_preview_area();
```

Comment:
I wonder if we can also remove these two lines because `compose_actions` is doing that same work anyway. I guess for the `clear_preview_area` call, I think `compose_clear_box_hooks` called via `clear_box` includes `clear_preview_area`, but that is only called if we didn't change the recipient -- I wonder if that reveals a latent bug where if we restore a draft into a compose box that already has some content there, if some parts of the `clear_box` process are incorrectly skipped. So the fix might be to change this:

```
    if (compose_state.composing() && !same_recipient_as_before(msg_type, opts)) {                       
        // Clear the compose box if the existing message is to a different recipient                    
        clear_box();                                                                                    
    }                                                                                                   
```

to be `compose_state.composing() && !same_recipient_as_before(msg_type, opts) && !opts.draft_id` and delete this line.


For `compose_fade.clear_compose`, I would expect that the `compose_fade.start_compose` call in `complete_starting_tasks` already does that, and we can safely just delete the call.

```
    compose_fade.clear_compose();                                                                       
    compose.clear_preview_area();                                                                       
```

We can then do a little commit moving `compose.empty_topic_placeholder()` to `compose_state.ts`, and have an end result where `drafts.ts` doesn't directly import `compose.js`, which possibly eliminates the major import cycle among the compose functions.

Reply:
Thanks for the suggestions! Implemented them after careful inspection of the code plus running the test suite.